src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -49,8 +50,24 @@ def create_configstore(client,
                                             sku=Sku(name=sku),
                                             tags=tags,
                                             public_network_access=public_network_access,
-                                            disable_local_auth=disable_local_auth)
+                                            disable_local_auth=disable_local_auth,
+                                            soft_delete_retention_in_days=retention_days,
+                                            enable_purge_protection=enable_purge_protection)
+
+    return client.begin_create(resource_group_name, name, configstore_params)
+
 
+def recover_deleted_configstore(cmd, client, name, resource_group_name=None, location=None, yes=False):
```

Comment:
>resource_group_name=None, location=None [](http://example.com/codeflow?start=51&length=39)

Add argument explanations in _params.py

Reply:
`location` and `resource_group_name` are global parameters and they come up in help as well.
Do we still want to include them in `_params.py`?

Reply:
For recover, the definition of location and resource group name should be a little different right? Like mentioning it is not configurable,  cannot recover to a new group/location.

Reply:
Correct. I updated the help text now.